stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/interceptor/internal/SendMessageInterceptorImpl.kt

```diff
@@ -173,19 +173,17 @@ internal class SendMessageInterceptorImpl(
      */
     private suspend fun uploadAttachments(message: Message, channelType: String, channelId: String): Result<Message> {
         return when {
-            globalState.isOnline() ->
-                if (message.hasPendingAttachments()) {
-                    waitForAttachmentsToBeSent(message, channelType, channelId)
-                } else {
-                    Result.success(message.copy(type = Message.TYPE_REGULAR))
-                }
+            globalState.isOnline() && message.hasPendingAttachments() ->
+                waitForAttachmentsToBeSent(message, channelType, channelId)
 
-            message.hasPendingAttachments() -> {
+            !globalState.isOnline() && message.hasPendingAttachments() -> {
                 // We enqueue attachments upload here if user is offline but an error is returned so message is not sent right away.
                 enqueueAttachmentUpload(message, channelType, channelId)
                 Result(ChatError("Chat is offline, not sending message with id ${message.id} and text ${message.text}"))
             }
 
+            !message.hasPendingAttachments() -> Result.success(message.copy(type = Message.TYPE_REGULAR))
+
```

Comment:
Should we check not only the WS connection but also the network availability then?
It seems like we should keep returning errors in the case when the mobile is offline.

Reply:
good call. I think this change is not so easy as we should separate socket and network state first

Reply:
I think it would be nice to check both cases, there's no reason to stop the message if only the socket is disconnected and there is a connection available. When both socket and network states are separated, I think we should do that. 

I refactored the code to avoid `if` inside the `when`, but the logic for the cases is the same, the only real addition is:
```
!message.hasPendingAttachments() -> Result.success(message.copy(type = Message.TYPE_REGULAR))
```

I think we should return `succcess` when there's no attachment to be sent. Otherwise, we are sending a failure for something that doesn't have to be done =|. If we return `success` we allow the flow of sending a message to continue, because it will be saved in the database and it will be sent if there's a connection because we will make a rest call in the `ChatClient`.

Reply:
But now checking, the `else` case should be removed in this code...

Reply:
as discussed on Slack: let's leave it as is and improve in the future after completing #3795